python/az/aro/azext_aro/_rbac.py

```diff
@@ -32,11 +33,8 @@ def assign_contributor_to_vnet(cli_ctx, vnet, object_id):
         type='roleDefinitions',
         name=DEVELOPMENT_CONTRIBUTOR if rp_mode_development() else CONTRIBUTOR,
     )
-
-    for assignment in list(client.role_assignments.list_for_scope(vnet)):
-        if assignment.role_definition_id.lower() == role_definition_id.lower() and \
-                assignment.principal_id.lower() == object_id.lower():
-            return
+    if has_assignment(list(client.role_assignments.list_for_scope(vnet)), role_definition_id, object_id):
```

Comment:
I wonder why list() is needed

Reply:
I believe it is because the result is a:
```
<azure.mgmt.authorization.v2018_09_01_preview.models.role_assignment_paged.RoleAssignmentPaged object at 0x7ff3e65b39d0>
```
This is an iterator object.  This could be easily be passed around but would require iteration before access to the objects inside.  For consistency sake, it is probably better to call `list()` and have all objects present before passing it around.  I followed the pattern done in the `assign_contributor_to_vnet` function.

Reply:
I think `has_assignment` will be fine accepting iterator: we loop through `assignments` there anyway. With `list(...)` there just an extra loop.

Reply:
Updated.